backend/benefit/calculator/models.py

```diff
@@ -0,0 +1,215 @@
+        verbose_name=_("other expenses"),
+        decimal_places=2,
+        max_digits=7,
+    )
+    start_date = models.DateField(verbose_name=_("benefit start from date"))
+    end_date = models.DateField(verbose_name=_("benefit end date"))
+    state_aid_max_percentage = models.IntegerField(
+        verbose_name=_("State aid maximum %"),
+        choices=STATE_AID_MAX_PERCENTAGE_CHOICES,
+    )
+
+    calculated_benefit_amount = models.DecimalField(
+        max_digits=7,
+        decimal_places=2,
+        verbose_name=_("amount of the benefit granted, calculated by the system"),
+        blank=True,
+        null=True,
+    )
+    override_benefit_amount = models.DecimalField(
+        max_digits=7,
+        decimal_places=2,
+        verbose_name=_(
+            "amount of the benefit manually entered by the application handler"
+        ),
+        blank=True,
+        null=True,
+    )
+
+    @property
+    def benefit_amount(self):
+        if self.override_benefit_amount is not None:
+            return self.override_benefit_amount
+        else:
+            return self.calculated_benefit_amount
+
+    override_benefit_amount_comment = models.CharField(
+        max_length=256,
+        verbose_name=_("reason for overriding the calculated benefit amount"),
+    )
+
+    history = HistoricalRecords(table_name="bf_calculator_calculator_history")
+
+    @property
+    def duration_in_months(self):
+        # The calculation calculation Excel file used the DAYS360 function, so we're doing the same
+        return duration_in_months(self.start_date, self.end_date)
+
+    def __str__(self):
+        return f"Calculation for {self.application}"
+
+    class Meta:
+        db_table = "bf_calculator_calculation"
+        verbose_name = _("calculation")
+        verbose_name_plural = _("calculations")
+        ordering = ("created_at",)
+
+
+class PaySubsidy(UUIDModel, TimeStampedModel):
+    """
+    Information about pay subsidies, as entered by the handlers in the calculator.
+    """
+
+    application = models.ForeignKey(
+        Application,
+        verbose_name=_("application"),
+        related_name="calculator_pay_subsidies",
+        on_delete=models.CASCADE,
+    )
+    start_date = models.DateField(verbose_name=_("Pay subsidy start date"))
+    end_date = models.DateField(verbose_name=_("Pay subsidy end date"))
+    amount = models.DecimalField(
+        max_digits=7, decimal_places=2, verbose_name=_("amount of the pay subsidy")
+    )
+    history = HistoricalRecords(table_name="bf_calculator_paysubsidy_history")
+
+    def __str__(self):
+        return f"PaySubsidy {self.start_date}-{self.end_date} of {self.amount}"
+
+    class Meta:
+        db_table = "bf_calculator_paysubsidy"
+        verbose_name = _("pay subsidy")
+        verbose_name_plural = _("pay subsidies")
+        # actual date range validation in serializer
+        unique_together = [("application", "start_date")]
```

Comment:
I suppose this is part of the business rules and there won't be any special cases where this might cause an issue?

Reply:
Good question, it turns out there's a special case for two pay subsidies overlapping, even starting on the same day